src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2976,10 +2999,18 @@ void MergeTreeData::checkAlterIsPossible(const AlterCommands & commands, Context
             }
         }
 
-        if (command.type == AlterCommand::MODIFY_ORDER_BY && !is_custom_partitioned)
+        if (command.type == AlterCommand::MODIFY_ORDER_BY)
         {
-            throw Exception(ErrorCodes::BAD_ARGUMENTS,
-                            "ALTER MODIFY ORDER BY is not supported for default-partitioned tables created with the old syntax");
+            bool allow_suspicious_indices = getSettingsRef()->allow_suspicious_indices;
```

Comment:
Doesn't compile for me.

Reply:
because it should be called on `local_context` my bad, but in any case at the start of function we have 
```
    const auto & settings = local_context->getSettingsRef();
    const auto & settings_from_storage = getSettings();
```
so why not just use those